Zend/zend_portability.h

```diff
@@ -350,7 +350,7 @@ char *alloca();
 # define ALLOCA_FLAG(name) \
 	bool name;
 # define SET_ALLOCA_FLAG(name) \
-	name = 1
+	name = true
 # define do_alloca_ex(size, limit, use_heap) \
 	((use_heap = (UNEXPECTED((size) > (limit)))) ? emalloc(size) : alloca(size))
 # define do_alloca(size, use_heap) \
```

Comment:
I'm wondering, would it make sense to not enable it in debug mode? Which would make it clearer that a memory leak can appear if one forgets the corresponding free. As currently only macOS on CI is our safeguard, and I still don't really understand why it's not enabled on macOS

Reply:
I don't know why it's not available on macOS, but the detection might not be handled properly from our side. Apparently, we doing [`AC_CHECK_HEADERS([alloca.h])`](https://github.com/php/php-src/blob/7e272b9a109ec13b774ec6983cd830f7951b1982/configure.ac#L389), but later also [`AC_FUNC_ALLOCA`](https://github.com/php/php-src/blob/7e272b9a109ec13b774ec6983cd830f7951b1982/configure.ac#L691), although [that macro is supposed to check for alloca.h and set `HAVE_ALLOCA_H`](https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Particular-Functions.html), too. That might not be a real problem, though (I don't know enough about autoconf). I guess someone with access to a macOS machine would need to debug.

Anyhow, it might be a good idea to disable `alloca()` for debug mode.

Reply:
line 348, if s guarded with `!defined(DARWIN)`

Reply:
seems it s like this since a long time, maybe the difference is no longer true ?